brew/src/main/java/nl/tudelft/serg/evosql/brew/generator/junit/JUnitGenerator.java

```diff
@@ -104,57 +116,17 @@ private void addConnectionDataFields(TypeSpec.Builder typeSpecBuilder) {
     }
 
     /**
-     * Generates a method specification for the runSQL method.
-     *
-     * @return method.
+     * Adds suitable run methods.
      */
-    private MethodSpec generateRunSQL() {
-        MethodSpec.Builder runSQL = MethodSpec.methodBuilder("runSQL")
-                .addModifiers(Modifier.PRIVATE, Modifier.STATIC)
-                .returns(TypeName.INT)
-                .addParameter(String.class, "sql")
-                .addParameter(boolean.class, "isUpdate");
-
+    private void addRunMethods(TypeSpec.Builder typeSpecBuilder) {
+        JUnitGeneratorHelper helper = new JUnitGeneratorHelper();
         if (jUnitGeneratorSettings.isGenerateSqlExecutorImplementation()) {
-            runSQL.addJavadoc(""
-                    + "Executes an SQL query on the database.\n\n"
-                    + "@param  query    The SQL query to execute\n"
-                    + "@param  isUpdate Whether the query is a data modification statement.\n"
-                    + "@return Whether the query execution has succeeded.\n")
-                    .beginControlFlow("try")
-                    .addStatement(
-                            "$T connection = $T.getConnection($L, $L, $L)",
-                            Connection.class, DriverManager.class,
-                            NAME_DB_JDBC_URL, NAME_DB_USER, NAME_DB_PASSWORD)
-                    .addStatement("$T statement = connection.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE, " +
-                            "ResultSet.CONCUR_READ_ONLY)", Statement.class)
-                    .beginControlFlow("if (isUpdate == true)")
-                    .addStatement("return statement.executeUpdate(sql)")
-                    .nextControlFlow("else")
-                    .addStatement("$T resultSet = statement.executeQuery(sql)", ResultSet.class)
-                    .beginControlFlow("if(resultSet.last())")
-                    .addStatement("return resultSet.getRow()")
-                    .nextControlFlow("else")
-                    .addStatement("return 0")
-                    .endControlFlow()
-                    .endControlFlow()
-                    .nextControlFlow("catch ($T sqlException)", SQLException.class)
-                    .addStatement("sqlException.printStackTrace()")
-                    .addStatement("return -1")
-                    .endControlFlow();
+            typeSpecBuilder.addMethod(helper.buildRunSqlImplementation());
+            typeSpecBuilder.addMethod(helper.buildGetResultColumns());
+            typeSpecBuilder.addMethod(helper.buildMapMaker());
```

Comment:
Shouldn't the mapmaker always be generated? Or somehow perhaps configured independently from the `generateSqlExecutorImplementation` parameter since they are perhaps less related?

Reply:
Hmm good one, it's a helper method for the `runSql` implementation, and since that method is the only caller to the map maker, I feel that it's justified to only generate it if the implementation is generated too. I do agree that it should be changed in the future if we need it for some other purpose (but I can't think of any right now).

Reply:
The only problem that is left is that some generated classes will not compile as there is not skeleton method as well if it isn't generated (generated test file: `JUnit4MediumTest.java`).

Reply:
Yeah you're absolutely correct, oops.